boards: mm_swiftio: fix sdcard power issue

lgl88911 opened the pull request:
In mm_swiftio hardware design, NVCC_SD0 of rt1052 is directly connected
to 3.3V, without using VSELECT control, so 1.8V support needs to be
disabled through software to avoid compatibility problems on UHS card.

Signed-off-by: Frank Li <lgl88911@163.com>

galak commented on subsys/disk/disk_access_usdhc.c:
```diff
@@ -618,8 +618,13 @@ enum usdhc_capability_flag {
 		USDHC_HOST_CTRL_CAP_VS30_MASK,
 	/*!< Support voltage 3.0V */
 	USDHC_SUPPORT_V180_FLAG =
+#if defined(CONFIG_BOARD_MM_SWIFTIO)
```
why the ifdef here?

lgl88911 replied:
In mm_swiftio hardware design, power of sdcard slot is directly connected to 3.3V and can't be switched to 1.8V by VSELECT, so the usdhc and SD card can only use 3.3V signal communication.In order to avoid compatibility problems caused by UHS card using 1.8v signal communication, we can only set usdhc not to support 1.8V on the software.

MaureenHelm replied:
We need to find another way to do this that doesn't require board-specific conditionals in the driver.

lgl88911 replied:
Do you means to use devicetree or Kconfig for configuration? devicetree looks more suitable, can you give some suggestions? Thanks

lgl88911 replied:
@MaureenHelm @galak I have changed to use Kconfig to control whether to disable 1.8v high-speed communication. Any board can enable this function according to the needs of the hardware. Please help review whether it is feasible. Thanks!

lgl88911 committed:
disk: sdhc: add option to disable 1.8v

When 1.8V is disabled, sdhc can only
communicate at low speed. But this can
save the external circuit for switching
between 3.3V and 1.8V, which is very
practical in costdown scenarios.

Signed-off-by: Frank Li <lgl88911@163.com>

lgl88911 committed:
boards: mm_swiftio: fix sdcard power issue

In mm_swiftio hardware design, NVCC_SD0 of rt1052 is directly connected
to 3.3V, without using VSELECT control, so 1.8V support needs to be
disabled through software to avoid compatibility problems on UHS card.

Signed-off-by: Frank Li <lgl88911@163.com>

lgl88911 commented:
@MaureenHelm @galak Can you give some suggestions for adding configuration options?  For the purpose of reducing the number of external components, I hope that the SDHC can always work under 3.3V in a scenario that does not require high-speed communication.

lgl88911 commented:
@MaureenHelm @galak This PR has been idle for a long time, can you give some advice on disable SDHC 1.8V communication. Thanks so much!

lgl88911 commented:
It seems that this PR has been forgotten, can you give some suggestions for improvement, Thank a lot!

MaureenHelm commented:
@JunYangNXP will you please have a look?

lgl88911 commented:
Since the usdhc code was moved to the driver, this pr was abolished and create https://github.com/zephyrproject-rtos/zephyr/pull/34205
